### PR TITLE
Fixed #33434 -- Changed font-size in admin CSS to use rem units.

### PR DESCRIPTION
**Pull request by rvikunwar**
In django most of the font size defination are absolute, in this PR I have updated font size unit (px) to rem which is and relative and responsive to <html> tag font size.
https://code.djangoproject.com/ticket/33434

### CODE REVIEW

File: django/contrib/admin/static/admin/css/base.css [hunk 1]
```diff
@@ -376,7 +376,7 @@ table thead th.sorted .sortoptions {
 }
 
 table thead th.sorted .sortpriority {
-    font-size: .8em;
+    font-size: .8rem;
```
Comment (kezabelle) [inline django/contrib/admin/static/admin/css/base.css:379]: With no opinion either way, I'm highlighting this line for eyeballing. Changing from `em` to `rem` here _may_ change the display to be larger/smaller than intended, based on the parent's font cascade; it's probably _fine_ and it is undeniably now more _consistent_ but it should be specifically checked and considered.

**Comment by claudep** [comment]
Having those fractional `0.625rem;`/`0.8125rem`/etc. scattered all over the CSS does not look so nice to me. I would be in favor of defining CSS variables instead, like `--font-14`, `--font-13`, etc. 

For information, here are the currently used sizes in pixels: 1, 9, 10, 11, 12, 13, 14, 16, 18, 20, 24
@rvikunwar, don't change anything until an agreement is found.

File: django/contrib/admin/static/admin/css/vendor/select2/select2.css [hunk 2]
```diff
@@ -41,7 +41,7 @@
     .select2-container .select2-search--inline .select2-search__field {
       box-sizing: border-box;
       border: none;
-      font-size: 100%;
+      font-size: 1rem;
```
Comment (claudep) [inline django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: I would not change this as it's a vendored CSS (and 100% looks OK to me).

Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: Thanks for your suggestion, I will change it back to 100%

Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: @claudep should i use  CSS variables or it is fine for now??

Comment (felixxm) [reply django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: We shouldn't change anything in a vendored CSS.

Comment (claudep) [reply django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: > @claudep should i use CSS variables or it is fine for now??

No, please wait for confirmation by others before doing that.

**Review by thibaudcolas — COMMENTED** [review]
Looks good to me overall, only a couple of things to change!

I have no opinions on the CSS variables refactoring – it would be nice, but it’s not really needed to solve the problem at hand. In my opinion the font-size definitions appearing messy is more of a reflection of the lack of consistency in the type scale (or the lack thereof). I’d recommend fixing that too, but it doesn’t have to be tied together with this PR.

File: django/contrib/admin/static/admin/css/base.css [hunk 3]
```diff
@@ -221,22 +221,22 @@ hr {
     border: none;
     margin: 0;
     padding: 0;
-    font-size: 1px;
+    font-size: 0.0625rem;
```
[the remark this answers is not in the document]
Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/base.css:224]: yes, it is not gonna chnage anything

**Comment by pauloxnet** [comment]
Can you add a link to the issue in the description of this PR? And in all PRs in future, please

**Comment by felixxm** [comment]
@rvikunwar Do you have time to keep working on this?

File: django/contrib/admin/static/admin/css/base.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/base.css:224]: @thibaudcolas should I remove this or this is fine?

File: django/contrib/admin/static/admin/css/base.css (hunk 1 above)
Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/base.css:379]: @ should I commit this one ??

File: django/contrib/admin/static/admin/css/base.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (thibaudcolas) [reply django/contrib/admin/static/admin/css/base.css:224]: This is fine – just wanted to check what you or others thought about it.

File: django/contrib/admin/static/admin/css/base.css (hunk 1 above)
Comment (thibaudcolas) [reply django/contrib/admin/static/admin/css/base.css:379]: Yes, we need this reverted back to `em`.

File: django/contrib/admin/static/admin/css/vendor/select2/select2.css (hunk 2 above)
Comment (thibaudcolas) [reply django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: @rvikunwar can you revert this?

```suggestion
      font-size: 100%;
```

**Review by thibaudcolas — CHANGES_REQUESTED** [review]
Almost there!

File: django/contrib/admin/static/admin/css/vendor/select2/select2.css (hunk 2 above)
Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/vendor/select2/select2.css:44]: @thibaudcolas I have already changed it back to 100% and also commited the change

File: django/contrib/admin/static/admin/css/base.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (thibaudcolas) [reply django/contrib/admin/static/admin/css/base.css:224]: I’d assume it doesn’t do anything but it’s just a guess – personally I wouldn’t feel comfortable changing this without testing, so would recommend keeping it in unless someone knows for sure / wants to spend the time testing.

[the remark this answers is not in the document]
Comment (claudep) [reply django/contrib/admin/static/admin/css/base.css:224]: This line was introduced in 2005 (dd5320d1d5)! I understand your safe attitude, but still I don't think it's healthy to keep lines for no obvious reason.

[the remark this answers is not in the document]
Comment (rvikunwar) [reply django/contrib/admin/static/admin/css/base.css:224]: what's the conclusion guys . .  need to wait for some time or should i commit this one ??

[the remark this answers is not in the document]
Comment (kezabelle) [reply django/contrib/admin/static/admin/css/base.css:224]: `font-size` **would** have an effect on an `<hr>` tag **if** the `margin` and `padding` properties weren't being set to `0`. 
An un-styled `<hr>` would have a responsive margin defined (e.g. `margin[...]: 0.5em`) and changing the font-size would change that margin.

In this case, I can't see that it is doing anything.

**Comment by felixxm** [comment]
@rvikunwar Thanks :+1: Welcome aboard :boat:

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**